lib/output/writer/http_client.go

```diff
@@ -109,12 +109,12 @@ func (h *HTTPClient) WriteWithContext(ctx context.Context, msg types.Message) er
 				parts[i] = msgCopy.Get(0)
 			}
 			parts[i].Set(p.Get())
-			if h.conf.CopyResponseHeaders {
```

Comment:
Unless I'm missing something, this check doesn't seem to be needed at all because the filtering is done in `internal/http/client.go` already.